iree/turbine/kernel/wave/compile.py

```diff
@@ -75,13 +77,19 @@ def invoke(self, *args, **kwargs):
             if usage == kernel_codegen.KernelBufferUsage.OUTPUT:
                 kernel_outputs.append(arg)
 
+        dynamic_symbols = []
+        for sym in self.options.dynamic_symbols:
+            arg, dim = self.symbols_args_map[sym]
+            dynamic_symbols.append(args[arg].shape[dim])
```

Comment:
Maybe list comprehension can be more efficient, but may not look as pretty:
```
dynamic_symbols = [args[self.symbols_args_map[arg_idx]].shape[self.symbols_args_map[dim]] for sym in self.options.dynamic_symbols]
```

Reply:
After the last fix this code became even more complex for the list comprehension (but I didn't noticed any measurable perf changes in e2e test)